File: 383. Ransom Note.py
class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        
        for x in ransomNote:
            if x in magazine:
                magazine=magazine.replace(x,"",1)
            else:
                return False
        return True

File: 383. Ransom Note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        int[] m = new int[26];
        for (int i=0; i<magazine.length(); i++){
            m[magazine.charAt(i)-'a']++;
        }
        int[] r = new int[26];
        for (int i=0; i<ransomNote.length(); i++){
            r[ransomNote.charAt(i)-'a']++;
        }

        for (int j=0; j<26; j++){
            if (m[j]<r[j]) return false;
        }
        return true;
    }
}
